buildpack/v05/encoderdecoder.go

```diff
@@ -0,0 +1,42 @@
+package v05
+
+import (
+	"os"
+
+	"github.com/BurntSushi/toml"
+
+	"github.com/buildpacks/lifecycle/api"
+	"github.com/buildpacks/lifecycle/buildpack/layertypes"
+)
+
+type EncoderDecoder05 struct {
+}
+
+func NewEncoderDecoder() *EncoderDecoder05 {
+	return &EncoderDecoder05{}
+}
+
+func (d *EncoderDecoder05) IsSupported(buildpackAPI string) bool {
+	return api.MustParse(buildpackAPI).Compare(api.MustParse("0.6")) < 0
+}
```

Comment:
Is it weird to call it `EncoderDecorder05` but have it work for APIs `0.2-0.5` maybe we want to create a package for each supported api? It would be a lot of duplication but it would be explicit and afterwards we could start pushing other API dependent logic into this packages?

We don't need to do this now but I wonder if our lives would be easier in the future if we just pushed everything into an api package and started a new API by copy paste the package and making the desired changes. The top level `buildpack` package could just just the buildpack API and delegate, so higher level workflows like `builder` wouldn't need to know about the api packages.

Reply:
I agree that calling it `EncoderDecoder05` while it's relevant for buildpack APIs `0.2-0.5` is very weird!
But I'm also not a fan of having a lot of duplicated code.
I think it really depends on what happens in the next versions. If there are many changes every version so probably it's worth having a package per version, but if there won't be any changes in the next few versions, maybe it's not worth having so much code duplicated for no reason.
I'll keep it as is for now and will add a comment that we need to work on it.

Reply:
Added a comment: https://github.com/buildpacks/lifecycle/pull/537/commits/10e7898473c8b663ac444985d8b6fe429f57b8ba